test/scriptFlywheel/unit/borrowRatePerBlockTests.spec.ts

```diff
@@ -0,0 +1,183 @@
+    let ONE_DAI: any;
+    let ONE_ETH: any;
+    let ONE_USDC: any;
+    // testEngine = new TestEngine();
+    // testEngine.deploy("scriptFlywheel.scen");
+
+    before(function () {
+        // Things to initialize before all test
+        this.timeout(0);
+        testEngine = new TestEngine();
+        // testEngine.deploy("scriptFlywheel.scen");
+    });
+
+    beforeEach(async function () {
+        this.timeout(0);
+        savingAccount = await testEngine.deploySavingAccount();
+        tokenInfoRegistry = await testEngine.tokenInfoRegistry;
+        accountsContract = await testEngine.accounts;
+        bankContract = await testEngine.bank;
+        // 1. initialization.
+        tokens = await testEngine.erc20Tokens;
+        mockChainlinkAggregators = await testEngine.mockChainlinkAggregators;
+        addressDAI = tokens[0];
+        addressUSDC = tokens[1];
+        addressMKR = tokens[4];
+        addressTUSD = tokens[3];
+        addressUSDT = tokens[2];
+
+        addressWBTC = tokens[8];
+
+        mockChainlinkAggregatorforTUSDAddress = mockChainlinkAggregators[3];
+        mockChainlinkAggregatorforDAIAddress = mockChainlinkAggregators[0];
+        mockChainlinkAggregatorforUSDCAddress = mockChainlinkAggregators[1];
+        mockChainlinkAggregatorforETHAddress = mockChainlinkAggregators[9];
+        erc20DAI = await ERC20.at(addressDAI);
+        erc20USDC = await ERC20.at(addressUSDC);
+        erc20MKR = await ERC20.at(addressMKR);
+        erc20TUSD = await ERC20.at(addressTUSD);
+        mockChainlinkAggregatorforDAI = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforDAIAddress
+        );
+        mockChainlinkAggregatorforUSDC = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforUSDCAddress
+        );
+        mockChainlinkAggregatorforETH = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforETHAddress
+        );
+        mockChainlinkAggregatorforTUSD = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforTUSDAddress
+        );
+        cDAI_addr = await testEngine.tokenInfoRegistry.getCToken(addressDAI);
+        cUSDC_addr = await testEngine.tokenInfoRegistry.getCToken(addressUSDC);
+        cUSDT_addr = await testEngine.tokenInfoRegistry.getCToken(addressUSDT);
+        cWBTC_addr = await testEngine.tokenInfoRegistry.getCToken(addressWBTC);
+        cDAI = await MockCToken.at(cDAI_addr);
+        cUSDC = await MockCToken.at(cUSDC_addr);
+        cUSDT = await MockCToken.at(cUSDT_addr);
+        cWBTC = await MockCToken.at(cWBTC_addr);
+        numOfToken = new BN(1000);
+        ONE_DAI = eighteenPrecision;
+        ONE_ETH = eighteenPrecision;
+        ONE_USDC = sixPrecision;
+
+        await savingAccount.fastForward(1000);
+    });
+
+    context("getBorrowRatePerBlock()", async () => {
+        context("with 18 decimal Token", async () => {
+            context("Compound supported tokens", async () => {
```

Comment:
I could see `Compound supported tokens` context three times. However, you should have this context only once, and rest all tests for different decimals should be the child of this contex.